lnwire/shutdown.go

```diff
@@ -0,0 +1,125 @@
+package lnwire
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/roasbeef/btcd/btcec"
+)
+
+// Shutdown is sent by either side in order to initiate the cooperative closure
+// of a channel. This message is sparse as both sides implicitly have the
+// information necessary to construct a transaction that will send the settled
+// funds of both parties to the final delivery addresses negotiated during the
+// funding workflow.
+//
+// NOTE: The requester is able to only send a signature to intiiate the
+// cooperative channel closure as all transactions are assembled observing BIP
+// 69, which defines a canonical ordering for inputs/outputs. Therefore, both
+// sides are able to arrive at an identical closure transaction as they know
+// the order of the inputs/outputs.
+type Shutdown struct {
+	// ChannelID serves to identify which channel is to be closed.
+	ChannelID ChannelID
+
+	// Len describes the length of the key that will be used to receive
```

Comment:
This is incorrect, the field of the shutdown struct (other than the target channel) should be either a `btcutil.Address` or a plain `[]byte`. Personally, I lean more towards using a `btcutil.Address` within the field as it allow us to a number of related utilities.